### PR TITLE
fix(ramda): nth function overload

### PR DESCRIPTION
**Pull request by songhuige**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [https://github.com/ramda/ramda/blob/master/source/nth.js](https://github.com/ramda/ramda/blob/master/source/nth.js)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ramda ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ramda@latest/))
was missing the following properties:
1. sequence
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 31af6c5328ad3b53588beeb94f931d3f2724fb41
</p>

File: types/ramda/test/nth-tests.ts [hunk 1]
```diff
@@ -2,8 +2,15 @@ import * as R from 'ramda';
 
 () => {
   const list = ['foo', 'bar', 'baz', 'quux'];
+  const str = 'abcd';
+
   R.nth(1, list); // => 'bar'
   R.nth(-1, list); // => 'quux'
   R.nth(-99, list); // => undefined
   R.nth(-99)(list); // => undefined
+  R.nth(1, str); // => 'b'
```
Comment (adispring) [inline types/ramda/test/nth-tests.ts:11]: Can you add `$ExpectType` to do type tests?

Use $ExpectType to assert that an expression is of a given type, and $ExpectError to assert that a compile error.

**Review by adispring — CHANGES_REQUESTED** [review]

File: types/ramda/test/nth-tests.ts (hunk 1 above)
Comment (Nicholaiii) [reply types/ramda/test/nth-tests.ts:11]: I agree with this sentiment. You don't need them for all of the tests you've written, but one that demonstrates each overload would be nice.

**Review by Nicholaiii — CHANGES_REQUESTED** [review]
Great addition, please check comments.

File: types/ramda/test/nth-tests.ts (hunk 1 above)
Comment (songhuige) [reply types/ramda/test/nth-tests.ts:29]: I modified the tests and added $ExpectType to each one

**Review by Nicholaiii — APPROVED** [review]
LGTM

File: types/ramda/index.d.ts [hunk 2]
```diff
@@ -1434,10 +1434,14 @@ export function none<T>(fn: (a: T) => boolean): (list: readonly T[]) => boolean;
...
 
 /**
- * Returns the nth element in a list.
- */
-export function nth<T>(n: number, list: readonly T[]): T | undefined;
-export function nth(n: number): <T>(list: readonly T[]) => T | undefined;
+ * Returns the nth element of the given list or string
+ */
+ export function nth<T>(n: number, list: readonly T[]): T | undefined;
+ export function nth(index: number, list: string): string;
```
Comment (adispring) [inline types/ramda/index.d.ts:1440]: LGTM.
One trivial request, can you change `index` to `n`, or change all `n` to `index`?

**Review by adispring — CHANGES_REQUESTED** [review]

**Review by adispring — APPROVED** [review]

**Review by Nicholaiii — APPROVED** [review]

**Comment by songhuige** [comment]
Ready to merge

**Pull request merged**